src/include/units/bits/common_quantity.h

```diff
@@ -90,34 +91,47 @@ using common_quantity_point_kind = std::common_type_t<QPK1, QPK2>;
 };
 
 template<units::QuantityPoint QP1, units::QuantityPointEquivalentTo<QP1> QP2>
-  requires requires { typename common_type_t<typename QP1::rep, typename QP2::rep>; }
-struct common_type<QP1, QP2> {
-  using type = units::quantity_point<
-    typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::dimension,
-    typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::unit,
-    typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::rep>;
+requires(requires { typename common_type_t<typename QP1::rep, typename QP2::rep>; } &&
+         std::is_same_v<typename QP1::origin, typename QP2::origin>) struct common_type<QP1, QP2> {
+  using type =
+      units::quantity_point<typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::dimension,
+                            typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::unit,
+                            typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::rep,
+                            typename QP1::origin>;
 };
 
 template<units::QuantityKind QK1, units::QuantityKindEquivalentTo<QK1> QK2>
-  requires requires { typename common_type_t<typename QK1::rep, typename QK2::rep>; }
+requires requires
+{
+  typename common_type_t<typename QK1::rep, typename QK2::rep>;
+}
 struct common_type<QK1, QK2> {
-  using type = units::quantity_kind<typename QK1::kind_type,
-    typename common_type_t<typename QK1::quantity_type, typename QK2::quantity_type>::unit,
-    typename common_type_t<typename QK1::quantity_type, typename QK2::quantity_type>::rep>;
+  using type =
+      units::quantity_kind<typename QK1::kind_type,
+                           typename common_type_t<typename QK1::quantity_type, typename QK2::quantity_type>::unit,
+                           typename common_type_t<typename QK1::quantity_type, typename QK2::quantity_type>::rep>;
 };
 
 template<units::QuantityPointKind QPK1, units::QuantityPointKindEquivalentTo<QPK1> QPK2>
-  requires requires { typename common_type_t<typename QPK1::rep, typename QPK2::rep>; }
+requires (requires
+{
+  typename common_type_t<typename QPK1::rep, typename QPK2::rep>;
+} && std::is_same_v<typename QPK1::origin, typename QPK2::origin>)
 struct common_type<QPK1, QPK2> {
-  using type = units::quantity_point_kind<typename QPK1::point_kind_type,
-    typename common_type_t<typename QPK1::quantity_kind_type, typename QPK2::quantity_kind_type>::unit,
-    typename common_type_t<typename QPK1::quantity_kind_type, typename QPK2::quantity_kind_type>::rep>;
+  using type = units::quantity_point_kind<
+      typename QPK1::point_kind_type,
+      typename common_type_t<typename QPK1::quantity_kind_type, typename QPK2::quantity_kind_type>::unit,
+      typename common_type_t<typename QPK1::quantity_kind_type, typename QPK2::quantity_kind_type>::rep,
+      typename QPK1::origin>;
 };
```

Comment:
In general, please do not mix modifications to the code with formatting because it is really hard to find what has changed. If we want to reformat the code we should do this in a separate commit/PR. But only when clang-format will learn how to do it properly.

Reply:
Yeah, I tried to avoid hitting cmd-B (in CLion for reformat) as I'm used to, but those lines it did automatically, while typing. I'll fix it.